### ISSUE TITLE
TLS flags do not work as documented

### ISSUE THREAD

**Issue by trevex** [vitessio/vitess#4287]
#### Overview of the Issue

The flags documented [here](https://vitess.io/user-guide/server-configuration/#db-config-parameters) do not work as intended, but rather some of the deprecated flags are still used throughout.

#### Reproduction Steps

Setting up TLS for replication:

1. On master tablet: `-db_flags 2048`
2. On repl tablet: `-db-config-repl-flags 2048`

The other documented flags, e.g. `db_repl_use_ssl`, do not work as intended, while the deprecated `db-config-repl-flags` does the trick.

#### Binary version

Fresh deployment using all `*:latest` docker images (no cache).

#### Operating system and Environment details

We ran into this on kubernetes, but it was reported by another user on Slack as well.

**Comment by msolters** [comment]
In MySQL there are several levels of SSL implementation:
*  No SSL at all
*  Enforce SSL to encrypt connections (can be per user with `REQUIRE SSL`), but not to authenticate or verify identity (only the SQL server should need cert, key, ca)
*  Enforce SSL contingent on specific properties of the client cert holding certain values.  E.g. check that the CN contains a certain substring etc.  This is used to not only encrypt but to verify _identity_.  In this case the client must provide a specific client cert.

As you can see, for SSL to work in all cases then the server must have a cert and key, but necessarily the client.  The logic of enabling e.g. `MASTER_SSL=1` when assembling `CHANGE MASTER` commands and so on should be fundamentally orthogonal to whether or not client certs are provided.

A suggestion is implementing a simpler `--ssl` flag that can be true or false to `vttablet`.  If the `vttablet` is a master, then it does what `-db_flags=2048` currently does.  If the `vttablet` is a replica, then it does what `-db-config-repl-flags=2048` currently does.  This simplifies the user interface and makes it simple to turn SSL on or off.

Providing certs should be conditionally orthogonal to this, which is to say we should be able to simply turn SSL on or off with a flag to `vttablet`, without providing or considering cert flags.  However, in the reverse direction, perhaps providing the cert flags should implicitly enable SSL.

**Comment by mattlord** [comment]
I'm closing this as stale. If there's still work to do here with Vitess v25 and MySQL 8.4+, let's please open a new ticket with those specifics. Thanks! ❤️
